packages/sync-actions/src/assets-actions.js

```diff
@@ -0,0 +1,41 @@
+import createBuildArrayActions, {
+  ADD_ACTIONS,
+  REMOVE_ACTIONS,
+  CHANGE_ACTIONS,
+} from './utils/create-build-array-actions'
+
+function toAssetIdentifier(asset) {
+  const assetIdentifier = asset.id
+    ? { assetId: asset.id }
+    : { assetKey: asset.key }
+  return assetIdentifier
+}
+
+export default function actionsMapAssets(diff, oldObj, newObj) {
+  const handler = createBuildArrayActions('assets', {
+    [ADD_ACTIONS]: newAsset => ({
+      action: 'addAsset',
+      asset: newAsset,
+    }),
+    [REMOVE_ACTIONS]: oldAsset => ({
+      action: 'removeAsset',
+      ...toAssetIdentifier(oldAsset),
+    }),
+    [CHANGE_ACTIONS]: (oldAsset, newAsset) => {
+      const result = []
+      result.push({
```

Comment:
Maybe adding a comment why we don't use `changeAsset` etc could help the next person to not wonder why?